homeassistant/components/schlage/lock.py

```diff
@@ -0,0 +1,78 @@
+"""Platform for Schlage lock integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from pyschlage.lock import Lock
+
+from homeassistant.components.lock import LockEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, MANUFACTURER
+from .coordinator import SchlageDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Schlage WiFi locks based on a config entry."""
+    coordinator: SchlageDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities(
+        SchlageLockEntity(coordinator=coordinator, device_id=device_id)
+        for device_id in coordinator.data.locks
+    )
+
+
+class SchlageLockEntity(CoordinatorEntity, LockEntity):
+    """Schlage lock entity."""
+
+    def __init__(
+        self, *, coordinator: SchlageDataUpdateCoordinator, device_id: str
+    ) -> None:
+        """Initialize a Schlage Lock."""
+        super().__init__(coordinator=coordinator)
+        self.device_id: str = device_id
+        self._attr_unique_id = device_id
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self.device_id)},
+            name=self._lock.name,
+            manufacturer=MANUFACTURER,
+            model=self._lock.model_name,
+            sw_version=self._lock.firmware_version,
+        )
+        self._update_attrs()
+
+    @property
+    def _lock(self) -> Lock:
+        """Fetch the Schlage lock from our coordinator."""
+        return self.coordinator.data.locks[self.device_id]
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._update_attrs()
+        return super()._handle_coordinator_update()
+
+    def _update_attrs(self) -> None:
+        """Update our internal state attributes."""
+        self._attr_name = self._lock.name
+        # When is_locked is None the lock is unavailable.
+        self._attr_available = self._lock.is_locked is not None
```

Comment:
This won't work as expected because `CoordinatorEntity` provides an `available` property which will take precedence.

You'll need to override the available property to be something like

```
@property
def available(self) -> bool:
   return super().available and self._lock.is_locked is not None
 ```